apps/api/v2/src/ee/bookings/inputs/create-booking.input.ts

```diff
@@ -0,0 +1,55 @@
+import { IsBoolean, IsNumber, IsString, IsOptional, IsArray } from "class-validator";
+
+import { TimeZone } from "@calcom/platform-constants";
+
+export class CreateBookingInput {
+  @IsString()
+  @IsOptional()
+  end?: string;
+
+  @IsString()
+  start!: string;
+
+  @IsNumber()
+  eventTypeId!: number;
+
+  @IsString()
+  @IsOptional()
+  eventTypeSlug?: string;
+
+  @IsString()
+  @IsOptional()
+  rescheduleUid?: string;
+
+  @IsString()
+  @IsOptional()
+  recurringEventId?: string;
+
+  @IsString()
+  timeZone!: TimeZone;
+
+  @IsOptional()
+  @IsArray()
+  @IsString({ each: true })
+  user?: string | string[];
+
+  @IsString()
+  language!: string;
+
+  @IsString()
+  @IsOptional()
+  bookingUid?: string;
+
+  metadata!: Record<string, string>;
```

Comment:
maybe we can improve this

Reply:
i guess only way is to gather all possible metadata keys and value types and create custom type